doc/zk-integration.md

```diff
@@ -0,0 +1,214 @@
+# Integration with ZK rollups
+
+Similarly to their optimistic alternatives, ZK rollups need to instantiate their derivation pipeline in order to
+integrate with Espresso. There are a number of alternatives for such integration, depending on specific constraints the
+rollup may have such as using a particular data availability layer in addition to Tiramisu, or the need to update their
+state at a faster pace than the Espresso light client contract does.
+
+In the following, we assume that the rollup relies on Tiramisu for data availability and explore two different
+integrations flavors. With integration 1, the rollup relies on the Espresso light client contract to fetch the latest
+Espresso state updates. while with integration 2 the rollup checks Espresso consensus directly inside its circuit.
+
+For both integrations:
+
+- We describe the high level structure of the circuit as well as the rollup L1 contract and how they are related.
+- We describe the use of an Espresso liveness escape hatch in the rollup L1 contract.
+
+When the escape hatch is activated, both integrations use the same circuit  
+ which corresponds to a mode where the rollup falls back to using its own (centralized) sequencer as shown in Figure 1.
+
+![image](zk-rollup-default-sequencer.svg)
+
+**Figure 1:** Circuit used when the rollup falls back to using its default (centralized) sequencer. The public inputs
+are the current state of the rollup _COMM_STATE_VM i_, the new rollup state after update _COMM_STATE_VM i+1_ and a
+commitment to the transactions applied to the state, _COMM_TXS_ROLLUP_. The private input (in bold) corresponds the list
+of transactions.
+
+## Integration 1: Rollup contract fetches Espresso block commitment from the Espresso light client contract
+
+For this integration, Espresso consensus verification is delegated to the Espresso light client contract. In practice
+the rollup contract will fetch the last Espresso block commitment and feed it to the circuit. Still additional gadgets
```

Comment:
Isn't the rollup block time much shorter than the interval to update the Espresso light client contract on the L1? My reading of Integration 1 is that rollups add blocks *more slowly than Espresso posts updates to the L1*, but that can't be what's intended.

Reply:
> Isn't the rollup block time much shorter than the interval to update the Espresso light client contract on the L1?

Yes.

> My reading of Integration 1 is that rollups add blocks more slowly than Espresso posts updates to the L1, but that can't be what's intended.

The section "Integration 1" describes the relation between the **rollup contract** and the circuit. The rollup indeed updates its state more frequently than Espresso posts updates on the L1. Is there a sentence in particular that generates this confusion? Should I add a note to highlight that the rollup contract is updated way less often than the rollup state itself?

Reply:
Added some sentence regarding the fact that rollup states are updated less frequently than the Espresso state. See e246dc999fc8c17fa98f25db7026e1945f1a3aa2.

Reply:
> Is there a sentence in particular that generates this confusion?

"In practice the rollup contract will fetch the last Espresso block commitment and feed it to the circuit."

Reads to me as though the rollup fetches its latest block (in real time) from Espresso. At the time I did not fully understand the design. Now I understand that proofs are not in real-time, but instead they are delayed according to what's been finalized on the L1. (Also, each proof batches *many* blocks, not just the latest block in real-time.)

> Should I add a note to highlight that the rollup contract is updated way less often than the rollup state itself?

Up to you. All else equal it's better to appeal to a wide range of readers, including those like me with poor understanding of the system. But all else is rarely equal, and often one needs to pick a minimum assumed reader understanding.

Reply:
Added a footnote in b8f3dcef45615cc2f5618f91cf2a8adf4ce0977d.